### PR TITLE
Obsolete outdated Visual Basic API

### PR DESCRIPTION
**Pull request by RussKie**
* `MsgBox` API is a legacy helper carried from VBA and Visual Basic era. It is essentially a wrapper over System.Windows.Forms.MessageBox API (which is itself is a wrapper over Win32 MessageBoxW API). Whilst the System.Windows.Forms.MessageBox API is being actively maintained and aligned with the Win32 API, the Visual Baisc API is neglected and fallen behind.

* `InputBox` API is also a legacy convinience API carried from the original Visual Basic into .NET Framework. It represents a form with an input text field. This API does not meet the required accessibility standards nor the modern Windows UI guidelines.

### CODE REVIEW

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (RussKie) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:548]: Do I need to apply the same to the actual implementations in here?
https://github.com/dotnet/runtime/blob/760a76bf4dc7d485f86c02dfe55713c75ee0121c/src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/Interaction.vb#L129-L135

If yes, we can't link Obsoletions.cs in that project, and we'll need to create a VB version of it...

**Comment by RussKie** [comment]
@merriemcgaw @KlausLoeffelmann FYI

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jkotas) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:57]: Undo these formatting changes. This is auto-generated file.

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jkotas) [reply src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:548]: > Do I need to apply the same to the actual implementations in here?

Yes.

You do not need to create a full blown VB version of Obsoletions.cs for this. It should be good enough to just add the attributes here with inlined strings.

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.csproj
Comment (jkotas) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.csproj:7]:
```suggestion
```
The ref files are auto-generated and have all strings inlined.

https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/updating-ref-source.md

**Comment by stephentoub** [comment]
Have these obsoletions been discussed in API review?

**Comment by danmoseley** [comment]
@RussKie I believe obsoleting API needs to go to API review for approval before making the change.

**Comment by danmoseley** [comment]
Marked for review

**Comment by danmoseley** [comment]
Setting milestone as 7.0 as I'm guessing that's your target? If not please change.

**Comment by RussKie** [comment]
I will send an email to the ARB. Thank you.

**Comment by teo-tsirpanis** [comment]
Removed blocking, this is targeting 8.0.0 and is not urgent, and [@bartonjs said that "it shouldn't be"](https://youtu.be/O-1AzEUXPTo?t=76).

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jeffhandley) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:57]:
```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Class, Inherited=false, AllowMultiple=false)]
```

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jeffhandley) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:490]:
```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Class, AllowMultiple=false, Inherited=false)]
```

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jeffhandley) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:588]:
```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Class, AllowMultiple=false, Inherited=false)]
```

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jeffhandley) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:729]:
```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Field, Inherited=false, AllowMultiple=false)]
```

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jeffhandley) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:737]:
```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Field, Inherited=false, AllowMultiple=false)]
```

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jeffhandley) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:889]:
```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Class, AllowMultiple=false, Inherited=false)]
```

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jeffhandley) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:1133]:
```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Parameter, Inherited=false, AllowMultiple=false)]
```

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jeffhandley) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:1139]:
```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Class, Inherited=false, AllowMultiple=false)]
```

File: src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs
Comment (jeffhandley) [inline src/libraries/Microsoft.VisualBasic.Core/ref/Microsoft.VisualBasic.Core.cs:1171]:
```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Class, Inherited=false, AllowMultiple=false)]
```

**Review by jeffhandley — APPROVED** [review]
Looks good to me once through API review.

**Review by KlausLoeffelmann — CHANGES_REQUESTED** [review]
Wait for pending API review.

**Comment by KlausLoeffelmann** [comment]
@RussKie: Assigned me and requested changes, so it doesn't get merged accidentally before the pending API review.

**Comment by RussKie** [comment]
Thank you @KlausLoeffelmann.

To reiterate the idea - the two API should be marked as obsoleted, and thus direct VB developers to use better alternatives. There are no plans or proposals to remove these API at any point.

**Comment by KlausLoeffelmann** [comment]
After further discussions in the context of the .NET runtime API Review, we have decided _not_ to take this PR.

"If the majority of the consumers are using these APIs from legacy code bases then obsoleting it is a disservice because it means upgrading to later versions of the product introduces additional work that they don’t care for. This might still make sense if there is something fundamentally wrong with the API, for example, if the APIs are insecure (BinaryFormatter comes to mind here). [The] understanding is that the primary consumer of these APIs are VB folks on legacy code bases and that the APIs aren’t insecure, just dated. “Dated” is a good description for most of the MS.VB APIs, but they have served that community well for about 20 years now. [...] We can just leave the APIs as they are and not evolve them."

Closing it.

**Pull request closed without merging**
